### DISCUSSION TITLE
How does fuzzy find work?

### DISCUSSION THREAD

**Discussion by brandoncc** [nvim-neo-tree/neo-tree.nvim#739] [Q&A] [answered]
I'm used to using Netrw, which I can search using `/` as a simple buffer. I don't understand why using `/` rarely seems to do anything in neo-tree, and even with it does something it always moves to something unexpected. I'm using astronvim and I'm not sure if that default config is making it act unexpectedly or if I just don't understand how to search the neo-tree buffer.

You can see in the video that I've attached that I am searching for "highlights" so I expected the cursor line to move to "highlights.lua", but it does nothing.

Thanks!

https://user-images.githubusercontent.com/543973/217621762-90d2b621-95bd-46a8-979f-f3065fd37857.mov

**Comment by brandoncc** [comment]
I also just found `filter_as_you_type` and tried to map that to `"/"`. The result is the same, I must be misunderstanding how it is supposed to work?

**Comment by miversen33** [comment, marked as answer]
Related: #721 

Fuzzy search is undergoing a massive overhaul so at this time it's probably not worth addressing this thread as whatever information is discussed/provided will be out of date soon. At least not until said overhaul finishes testing and is deployed (I'd guess that will be in a few weeks, depending on what bugs are found during the testing phase).

**Reply by brandoncc** [reply]
Does that mean there is literally no way to search the neotree buffer right now? Given that an existing implementation is being overhauled, I can't imagine that is true. Should I open a new discussion named "how do I search the tree and move the cursor to the first match"? That's all I want to know.

Thanks

**Reply by miversen33** [reply]
Fuzzy search != filter (though I understand how it can get confusing).

You can "search" your tree with either `/` (which does a search that is "called" fuzzy search, though its not as explained in #721 ), or `f`.

I have a feeling `f` (named "filter_on_submit") is what you are looking for.

**Reply by brandoncc** [reply]
Thanks, I understand the difference at a technical level. I called it fuzzy find because of the [command name in the default mappings](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/v2.x/lua/neo-tree/defaults.lua#L359), sorry for the confusion.

I have tried `f`, `/`, `D`, and changing the command for `"/"` to `filter_as_you_type`. All of these have the exact same result....they do nothing! By any chance, did you happen to watch the video I included? I just can't imagine that what I'm experiencing is the intended experience.

**Reply by brandoncc** [reply]
Oh....I know what was happening now! Before I found `f` and `D`, and then posted this question, I found [this issue](https://github.com/nvim-neo-tree/neo-tree.nvim/issues/354) and added this config, misunderstanding its purpose:

```lua
    find_args = {  -- you can specify extra args to pass to the find command.
      fd = {
        "--max-depth", "1",
      },
      find = {
        "-maxdepth", 0
      }
    },
```

I just removed this config and `/`/`f` are working now. `D` still doesn't seem to do what I would expect, but this is much better. I think I must have been moving too fast so `/` didn't seem like it was doing anything before....thank you for the help!

**Reply by brandoncc** [reply]
Once I give `D` a little time, it also works. The "search as you type" is just slow (probably a debounce timeout, I'm guessing).

**Reply by miversen33** [reply]
Awesome! I'm glad you got it worked out :) Keep an eye on that PR, that should add much better fuzzy finding once its merged
